lib/facter/consul_version.rb

```diff
@@ -0,0 +1,15 @@
+# consul_version.rb
+
+Facter.add(:consul_version) do
+  setcode do
+    if Facter.value(:kernel) == 'Linux'
+      version = Facter::Util::Resolution.exec('consul --version')
+      version = version.lines.first.split[1].tr('v','')
+    elsif Facter.value(:kernel) == 'windows'
+      version = Facter::Util::Resolution.exec('"C:\\Program Files\\Consul\\consul.exe" --version')
```

Comment:
Is there anyway to make this configurable as we can't be certain that this is actually where consul is installed on Windows.  For instance we install to C:\Consul not Program Files.